packages/cli/src/commands/help.ts

```diff
@@ -0,0 +1,213 @@
+import chalk from 'chalk';
+import stripAnsi from 'strip-ansi';
+import { LOGO, NAME } from '@vercel-internals/constants';
+
+const INDENT = ' '.repeat(2);
+const NEWLINE = '\n';
+
+export interface CommandOption {
+  name: string;
+  shorthand: string | null;
+  type: 'boolean' | 'string';
+  argument?: string;
+  deprecated: boolean;
+  description?: string;
+  multi: boolean;
+}
+export interface CommandArgument {
+  name: string;
+  required: boolean;
+}
+export interface CommandExample {
+  name: string;
+  value: string | string[];
+}
+export interface Command {
+  name: string;
+  description: string;
+  arguments: CommandArgument[];
+  options: CommandOption[];
+  examples: CommandExample[];
+}
+
+export function calcLineLength(line: string[]) {
+  return stripAnsi(lineToString(line)).length;
+}
+
+// Insert spaces in between non-whitespace items only
+export function lineToString(line: string[]) {
+  let string = '';
+  for (let i = 0; i < line.length; i++) {
+    if (i === line.length - 1) {
+      string += line[i];
+    } else {
+      const curr = line[i];
+      const next = line[i + 1];
+      string += curr;
+      if (curr.trim() !== '' && next.trim() !== '') {
+        string += ' ';
+      }
+    }
+  }
+  return string;
+}
+
+export function outputArrayToString(outputArray: string[]) {
+  return outputArray.join(NEWLINE);
+}
+
+/**
+ * Example: `▲ vercel deploy [path] [options]`
+ * @param command
+ * @returns
+ */
+export function buildCommandSynopsisLine(command: Command) {
+  const line: string[] = [LOGO, chalk.bold(NAME), chalk.bold(command.name)];
+  if (command.arguments.length > 0) {
+    for (const argument of command.arguments) {
+      line.push(argument.required ? argument.name : `[${argument.name}]`);
+    }
+  }
+  if (command.options.length > 0) {
+    line.push('[options]');
+  }
+  return lineToString(line);
+}
+
+export function buildCommandOptionLines(command: Command) {
+  const MAX_LINE_LENGTH = process.stdout.columns;
```

Comment:
Should be `client.stdout`

Reply:
Good callout. Noticed that the client types don't use `tty.WriteStream`, so I change that. Not sure if you did that intentionally or if you just didn't know that type existed. This is the type that `process.stderr` uses.

Reply:
Ya we should probably use that type then.